### PR TITLE
refactor: add type inference for scalar udfs

### PR DESCRIPTION
**Pull request by kevinbheda**

### CODE REVIEW

File: dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/ByteToString.java [hunk 1]
```diff
@@ -13,4 +28,33 @@
     public String eval(ByteString byteField) {
         return byteField.toStringUtf8();
     }
+
+    @Override
+    public TypeInference getTypeInference(DataTypeFactory typeFactory) {
```
Comment (arujit) [inline dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/ByteToString.java:33]: test for `getTypeInference` method?

File: dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/ByteToString.java [hunk 2]
```diff
@@ -13,4 +28,33 @@
...
+        TypeInference build = TypeInference.newBuilder()
+                .inputTypeStrategy(new InputTypeStrategy() {
+                    @Override
```
Comment (arujit) [inline dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/ByteToString.java:36]: Does it make sense to create a separate class for the `InputTypeStrategy` class? Would have made unit testing easier.

File: dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/MapGet.java [hunk 3]
```diff
@@ -1,6 +1,20 @@
 package io.odpf.dagger.functions.udfs.scalar;
```
[the remark this answers is not in the document]
Comment (kevinbheda) [reply dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/MapGet.java:1]: did this fix

File: dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/ByteToString.java (hunk 2 above)
Comment (kevinbheda) [reply dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/ByteToString.java:36]: yes

**Review by arujit — APPROVED** [review]
changes look fine to me. merging the changes.

**Pull request merged**